### PR TITLE
Per message config for elasticdump

### LINKED ISSUE #1868: Modify the elastic dump container to support saving snapshots from different indexes
Pulled out of #1778 as a nice-to-have but not essential feature of snapshots.

(Especially if the snapshot convertor can do V1 and V2 from the same index.)

### PR DESCRIPTION
**Pull request by kenoir**
### What is this PR trying to achieve?

Configure elasticdump on a per message basis (fixes #1868 )

### Who is this change for?

☄️ 💻 

### Have the following been considered/are they needed?

- [ ] Deployed new versions
- [ ] Run `terraform apply`.

### CODE REVIEW

File: data_api/elasticdump/run_elasticdump.py [hunk 1]
```diff
@@ -73,14 +90,11 @@ def run():
 
     print('*** Reading messages from SQS')
     message = get_message(sqs_client=sqs_client, sqs_queue_url=sqs_queue_url)
+    snapshot_request = parse_message(message)
```
Comment (alexwlchan) [inline data_api/elasticdump/run_elasticdump.py]: We should log the request and/or the message we received.

File: data_api/snapshot_scheduler/src/test_snapshot_scheduler.py [hunk 2]
```diff
@@ -15,10 +15,20 @@ def utcnow(cls):
 
 @mock.patch('datetime.datetime', patched_datetime)
 def test_writes_message_to_sqs(sns_client, topic_arn):
-    snapshot_scheduler.main(sns_client=sns_client)
+    target_bucket_name="target_bucket_name"
+    es_index="es_index"
```
Comment (alexwlchan) [inline data_api/snapshot_scheduler/src/test_snapshot_scheduler.py]: nit: Python convention is to have spaces around assignment operators.

File: data_api/snapshot_scheduler/src/test_snapshot_scheduler.py [hunk 3]
```diff
@@ -15,10 +15,20 @@ def utcnow(cls):
...
+
+    snapshot_scheduler._run(
```
Comment (alexwlchan) [inline data_api/snapshot_scheduler/src/test_snapshot_scheduler.py]: Ick. Underscore methods are generally treated as “private” – you can access them, but they’re treated as internal and you shouldn’t rely on them. Better would be to set up the environment variables you expect, and call the `main` function.

**Comment by kenoir** [comment]
DO NOT MERG! FLAKY TESTS DETECTED!

File: data_api/elasticdump/run_elasticdump.py [hunk 4]
```diff
@@ -28,6 +39,12 @@ def get_message(sqs_client, sqs_queue_url):
         return message
 
 
+def parse_message(message):
+    raw_messsage_dict = json.loads(message['Body'])
```
Comment (alexwlchan) [inline data_api/elasticdump/run_elasticdump.py]: messsage?

File: data_api/elasticdump/test_run_elasticdump.py [hunk 5]
```diff
@@ -12,25 +13,25 @@
...
-        'index_with_no_scheme',
-        'https://elastic:changeme@example.org:9200/index_with_no_scheme'
+            {
+                'es_username': 'elastic',
+                'es_password': 'changeme',
+                'es_hostname': 'example.org',
+                'es_port': 9200,
+            },
+            'index_with_no_scheme',
+            'https://elastic:changeme@example.org:9200/index_with_no_scheme'
```
Comment (alexwlchan) [inline data_api/elasticdump/test_run_elasticdump.py]: Does this extra indent add anything except cluttering the diff?

File: data_api/snapshot_scheduler/src/test_snapshot_scheduler.py [hunk 6]
```diff
@@ -15,10 +15,20 @@ def utcnow(cls):
 
 @mock.patch('datetime.datetime', patched_datetime)
 def test_writes_message_to_sqs(sns_client, topic_arn):
-    snapshot_scheduler.main(sns_client=sns_client)
+    target_bucket_name = "target_bucket_name"
+    es_index = "es_index"
+
+    snapshot_scheduler._run(
```
Comment (alexwlchan) [inline data_api/snapshot_scheduler/src/test_snapshot_scheduler.py]: See previous comment about testing with `_run`, come chat to me if you want to go over this.

File: data_api/elasticdump/test_run_elasticdump.py (hunk 5 above)
Comment (kenoir) [reply data_api/elasticdump/test_run_elasticdump.py]: This is how intellij decided to format the file, will fix.

**Pull request merged**